### PR TITLE
Upmerge openthread

### PR DESCRIPTION
**Pull request by SebastianBoe**
Update OpenThread to the latest changes.

OpenThread is now compatible with mbedtls 2 and 3.

This PR uses commits from and deprecates https://github.com/zephyrproject-rtos/zephyr/pull/38497

### CODE REVIEW

**Review by rlubos — APPROVED** [review]

File: west.yml [hunk 1]
```diff
@@ -183,7 +183,7 @@ manifest:
       revision: 6010f0523cbc75f551d9256cf782f173177acdef
       path: modules/lib/open-amp
     - name: openthread
-      revision: 5d706547ebcb0a85e11412bcd88e80e2af98c74d
+      revision: pull/52/head
```
Comment (rlubos) [inline west.yml:186]: Please update SHA: `98921015705c29ed283b3c696e33235dff419467`

Comment (SebastianBoe) [reply west.yml]: Done

**Review by rlubos — APPROVED** [review]

**Review by carlescufi — APPROVED** [review]

**Review by LuDuda — APPROVED** [review]

**Comment by SebastianBoe** [comment]
pushed whitespace to re-run CI.

**Comment by carlescufi** [comment]
@SebastianBoe replace with SHA?

**Comment by SebastianBoe** [comment]
I can't https://github.com/zephyrproject-rtos/openthread/pull/53 is not merged yet.

We made a mistake earlier, resulting in two upmerge PRs for this one manifest update.

File: west.yml [hunk 2]
```diff
@@ -183,7 +183,7 @@ manifest:
       revision: 6010f0523cbc75f551d9256cf782f173177acdef
       path: modules/lib/open-amp
     - name: openthread
-      revision: 5d706547ebcb0a85e11412bcd88e80e2af98c74d
+      revision: pull/53/head
```
Comment (rlubos) [inline west.yml:186]: Please update SHA: `fd1ab161f7d5cfd5d0120cd5d1797811e78ff918`

**Review by rlubos — APPROVED** [review]

**Pull request merged**